plugins/modules/redhat_subscription.py

```diff
@@ -547,7 +547,7 @@ def str2int(s, default=0):
         # - with subscription-manager < 1.26.5-1 (in RHEL < 8.2);
         #   fixed later by https://github.com/candlepin/subscription-manager/pull/2175
         # - sporadically: https://bugzilla.redhat.com/show_bug.cgi?id=2049296
-        if distro_id == 'fedora' or distro_version[0] >= 7:
+        if distro_id == 'fedora' or (distro_version[0] >= 7 and distro_version[1] > 3):
```

Comment:
note that this version comparison is not correct: in case the version is e.g. 8.1, then
- `distro_version[0] >= 7` -> `true`
- `distro_version[1] > 3` -> `false`

resulting in `false`, since the condition is `and`